packages/react-native-jigsaw/src/components/ToggleButton.tsx

```diff
@@ -0,0 +1,139 @@
+import * as React from "react";
+import { withTheme } from "../core/theming";
+import themeT from "../styles/DefaultTheme";
+import { colorTypes } from "../types";
+import { StyleProp, StyleSheet, ViewStyle } from "react-native";
+import IconButton from "./IconButton";
+import {
+  COMPONENT_TYPES,
+  GROUPS,
+  createIconProp,
+  FORM_TYPES,
+  PROP_TYPES,
+  createBoolProp,
+  createTextProp,
+  createColorProp,
+  createNumberProp,
+  createActionProp,
+  FIELD_NAME,
+} from "../core/component-types";
+
+type Props = {
+  icon: string;
+  status?: "checked" | "unchecked";
+  onPress?: () => void;
+  disabled?: boolean;
+  color?: colorTypes;
+  colorSecondary?: colorTypes;
+  borderColor?: colorTypes;
+  iconSize?: number;
+  width?: number;
+  height?: number;
+  style?: StyleProp<ViewStyle>;
+  accessibilityLabel?: string;
+  theme: typeof themeT;
+};
+
+const ToggleButton: React.FC<Props> = ({
+  icon,
+  status = "unchecked",
+  onPress = () => {},
+  disabled = false,
+  color = "primary",
+  colorSecondary = "surface",
+  borderColor = "divider",
+  iconSize = 25,
+  width = 50,
+  height = 50,
+  style,
+  accessibilityLabel = "toggle button",
+  theme: { colors },
+}) => {
+  return (
+    <IconButton
+      icon={icon}
+      size={iconSize}
+      color={status === "unchecked" ? colors[colorSecondary] : colors[color]}
+      onPress={onPress}
+      disabled={disabled}
+      accessibilityLabel={accessibilityLabel}
```

Comment:
you can remove these 3 accessibility props because an IconButton uses the new Pressable api which doesn't support these